File: src-tauri/src/handlers/patient.rs
use crate::datastore::patient;
use crate::datastore::search;
use crate::types::Patient;

#[tauri::command]
pub fn add_patient(data: Patient, window: tauri::Window) -> Result<String, String> {
    let id = patient::add_patient_db(data, window);
    println!("id: {:?}", id);
    id
}

#[tauri::command]
pub fn get_patient_by_id(patient_id: String, window: tauri::Window) -> Result<Patient, String> {
    let patient_res = patient::get_patient_by_id_db(patient_id, window);
    println!("patient_res: {:?}", patient_res);
    patient_res
}

#[tauri::command]
pub fn update_patient(id: String, data: Patient, window: tauri::Window) -> Result<String, String> {
    patient::update_patient_db(id, data, window)
}

#[tauri::command]
pub fn delete_patient(id: String, window: tauri::Window) -> Result<String, String> {
    patient::delete_patient_db(id, window)
}

#[tauri::command]
pub fn search_patients(search_term: String, window: tauri::Window) -> Result<Vec<Patient>, String> {
    print!("search_term: {:?}", search_term);
    search::search_patients_db(search_term, window)
}

File: src/api/patient.ts
import { invoke } from "@tauri-apps/api/core";
import { TPatientInfo } from "../types";
type TPatient = {
  id: string;
  name: string;
};
export const addPatientApi = async (data: TPatientInfo) => {
  try {
    const res = await invoke<string>("add_patient", { data: data });
    return res;
  } catch (e) {
    console.log(e);
  }
};
export const getPatientInfoByIdApi = async (id: string) => {
  if (!id) return;
  try {
    const res = await invoke<TPatientInfo>("get_patient_by_id", {
      patientId: id,
    });
    return res;
  } catch (e) {
    console.log(e);
  }
};
export const searchPatientApi = async (input: string) => {
  try {
    const res = await invoke<TPatient[]>("search_patients", {
      searchTerm: input,
    });
    return res;
  } catch (e) {
    console.log("Search error", e);
  }
};
export const updatePatientApi = async (id: string, patient: TPatientInfo) => {
  try {
    const res = await invoke("update_patient", {
      id,
      data: patient,
    });
    return res;
  } catch (e) {
    console.log(e);
  }
};
